internal/storage/v1/elasticsearch/factory_v1.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package elasticsearch
+
+import (
+	"context"
+	"flag"
+	"io"
+
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/internal/metrics"
+	"github.com/jaegertracing/jaeger/internal/storage/elasticsearch/config"
+	"github.com/jaegertracing/jaeger/internal/storage/v1"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/api/dependencystore"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/api/samplingstore"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/api/spanstore"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/api/spanstore/spanstoremetrics"
+	esDepStorev1 "github.com/jaegertracing/jaeger/internal/storage/v1/elasticsearch/dependencystore"
+	esSpanStore "github.com/jaegertracing/jaeger/internal/storage/v1/elasticsearch/spanstore"
+)
+
+var ( // interface comformance checks
+	_ storage.Factory        = (*Factory)(nil)
+	_ io.Closer              = (*Factory)(nil)
+	_ storage.Configurable   = (*Factory)(nil)
+	_ storage.Inheritable    = (*Factory)(nil)
+	_ storage.Purger         = (*Factory)(nil)
+	_ storage.ArchiveCapable = (*Factory)(nil)
+)
+
+type Factory struct {
+	Options     *Options
+	coreFactory *FactoryBase
+}
+
+func NewFactory() *Factory {
+	return &Factory{
+		Options: NewOptions(primaryNamespace),
+	}
+}
+
+func NewArchiveFactory() *Factory {
+	return &Factory{
+		Options: NewOptions(archiveNamespace),
+	}
+}
+
+func (f *Factory) AddFlags(flagSet *flag.FlagSet) {
+	f.Options.AddFlags(flagSet)
+}
+
+func (f *Factory) InitFromViper(v *viper.Viper, _ *zap.Logger) {
+	f.Options.InitFromViper(v)
+}
+
+func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
+	cfg := f.Options.GetConfig()
+	if err := cfg.Validate(); err != nil {
+		return err
+	}
+	defaultConfig := DefaultConfig()
+	cfg.ApplyDefaults(&defaultConfig)
+	if f.Options.Config.namespace == archiveNamespace {
+		aliasSuffix := "archive"
+		if cfg.UseReadWriteAliases {
+			cfg.ReadAliasSuffix = aliasSuffix + "-read"
+			cfg.WriteAliasSuffix = aliasSuffix + "-write"
+		} else {
+			cfg.ReadAliasSuffix = aliasSuffix
+			cfg.WriteAliasSuffix = aliasSuffix
+		}
+		cfg.UseReadWriteAliases = true
+	}
+	coreFactory, err := NewFactoryBase(*cfg, metricsFactory, logger)
+	if err != nil {
+		return err
+	}
+	f.coreFactory = coreFactory
+	return nil
+}
+
+// CreateSpanReader implements storage.Factory
+func (f *Factory) CreateSpanReader() (spanstore.Reader, error) {
+	params, err := f.coreFactory.GetSpanReaderParams()
+	if err != nil {
+		return nil, err
+	}
+	sr := esSpanStore.NewSpanReaderV1(params)
+	return spanstoremetrics.NewReaderDecorator(sr, f.coreFactory.GetMetricsFactory()), nil
```

Comment:
Factory is who passes metrics into base factory in the first place. It does not need a way to get it back.